File: tech-studio-projects/financial-industry/algo-trading-platform/cpp/orders/Rejection.cpp
#include "Rejection.h"

Rejection::Rejection(const IOrderCore& rejectedOrder, RejectionReason rejectionReason)
    : _rejectedOrder(rejectedOrder), _rejectionReason(rejectionReason) {}

long Rejection::GetOrderId() const {
    return _rejectedOrder.GetOrderId();
}

long Rejection::GetUsername() const {
    return _rejectedOrder.GetUsername();
}

long Rejection::GetSecurityId() const {
    return _rejectedOrder.GetSecurityId();
}

RejectionReason Rejection::GetRejectionReason() const {
    return _rejectionReason;
}

File: tech-studio-projects/financial-industry/algo-trading-platform/java/algo-trading-platform/src/com/fintech/algotrading/orders/Rejection.java
package com.fintech.algotrading.orders;

public class Rejection {
    private final IOrderCore rejectedOrder;
    private final RejectionReason rejectionReason;

    // Constructor initializes the Rejection with the order and reason.
    public Rejection(IOrderCore rejectedOrder, RejectionReason rejectionReason) {
        this.rejectedOrder = rejectedOrder;
        this.rejectionReason = rejectionReason;
    }

    // Getter for the order ID.
    public long getOrderId() {
        return rejectedOrder.getOrderId();
    }

    // Getter for the username.
    public long getUsername() {  // Assuming username is a String, adjust the type if necessary
        return rejectedOrder.getUsername();
    }

    // Getter for the security ID.
    public long getSecurityId() {
        return rejectedOrder.getSecurityId();
    }

    // Getter for the rejection reason.
    public RejectionReason getRejectionReason() {
        return rejectionReason;
    }
}
